### PR TITLE
Add Trino 386 release notes

### LINKED ISSUE #12748: Release notes for 386

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 386 release.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Docs only.

> How would you describe this change to a non-technical end user or system administrator?

Add the release notes for Trino 386 with sections for general, security, connectors and other topics as included in the release.

## Related issues, pull requests, and links

Fixes #12748

## Documentation

( ) No documentation is needed.
(✅) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅) No release notes entries required.
( ) Release notes entries required with the following suggested text:

## Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 9 June 2022 ✅ 

b5fdcf78d2ab830f79b621cbdca8c1c7f92d1896 @arhimondr #12754 ✅ rn ✅ docs
b4c9572bfdc55f646362a7ad96df66b4a5a7b37f @arhimondr #12754 ✅ rn ✅ docs
de2943beae839404f97233ec0c3f1405fb5d5d54 @arhimondr #12754 ✅ rn ✅ docs
cb75ca5e8696b6b6767d195937f55c974f6aa9ac @losipiuk #12733 ✅ rn ✅ docs
36944a2121b92a88e97eec31f22c3f1f8011fdb2 @ebyhr #12741 ✅ rn ✅ docs
237954bd473e21244d918acbad73b9d0fcf28b1a @ebyhr #12492 ✅ rn ✅ docs
1097006e27ecee79e9bdbc2888b74a189023955b @findepi #12672 ✅ rn ✅ docs
8bede1052c63f85da447f926888ae8f973763bdb @findepi #12672 ✅ rn ✅ docs
82960b70a551f06d36ce3c6eb5ac9d6af8134ce0 @findepi #12749 ✅ rn ✅ docs
70b9ac37875b9f62bc034803c94fd4ea6245d096 @findepi #12749 ✅ rn ✅ docs
10915601abfca4242901d63e88230484c285879e @findepi #12718 ✅ rn ✅ docs
b6703a0a9c2f1b696d44aad97a4bec82f4c37ca0 @findepi #12718 ✅ rn ✅ docs
cb78e32085fc5927bfc39d879ea0de2b5f2edfc0 @findepi #12718 ✅ rn ✅ docs
f9022aa918af20486c3b592af09753bfb3bce3da @findepi #12719 ✅ rn ✅ docs
45a84f54f94a154eea97cbf2f768964876bba15f @losipiuk #12746 ✅ rn ✅ docs
0c853ff2bec36af08e18766774910efb9ebe9ea6 @findepi #12612 ✅ rn ✅ docs
99ff1c6077cd219afaa10541c7bbfca03d7e02a2 @arhimondr #12646 ✅ rn ✅ docs
dbebd79013b561ffc82bc2da85872bd32fe178fd @electrum #12560 ✅ rn ✅ docs
fd59282be141d21a892d331b5b525c7f782769e1 @findepi #12767 ✅ rn ✅ docs
61c0e4e83e115eae030d9c997888325940764c57 @findepi #12767 ✅ rn ✅ docs
3f9877fc4ab79fec792ec65d6f2bb1abe83f1729 @findepi #12767 ✅ rn ✅ docs
cd09f28c6a21b828535347994c31e691c4b22baa @findepi #12489 ✅ rn ✅ docs
8c162f2aafb35fa254706b26c2b62084bcc12fb2 @findepi #12489 ✅ rn ✅ docs
9059319077a8ed7fc54b482e093e05f47da75f4d @findepi #12489 ✅ rn ✅ docs
bd6becaa29854078908b6636dcb5cbb95f026e29 @findepi #12687 ✅ rn ✅ docs
397522e6739f41a269600a90f817f5468d2c1ea1 @findepi #12687 ✅ rn ✅ docs
f793d7d2a3be7459f643cf44aae07a3127537f8c @findepi #12687 ✅ rn ✅ docs
4753ecc09f0224df122fc8182f0df508b344d087 @findepi #12687 ✅ rn ✅ docs
9608658731b6a9c85e3868ae22a6f07b2d5a9956 @findepi #12687 ✅ rn ✅ docs
8ebb4bd7d442d03d5d0049eb7a25099ff49ce1c9 @electrum #12750 ✅ rn ✅ docs
e75ba00ccb9f6cab967565ec34cacad8450fc5eb @findepi #12412 ✅ rn ✅ docs
fa856c4de1f18fab8015b8eb94de0b10a88264ce @ebyhr #12744 ✅ rn ✅ docs
88d8a0d436a8da0739d806b4caf54661a01a3828 @ebyhr #12752 ✅ rn ✅ docs
7c516e594eee7f1814b172b7fedf8c6cafd70038 @ebyhr #12764 ✅ rn ✅ docs
41d4f25e323568000837ce5d227e9a92a92df1bd @dain #12713 ✅ rn ✅ docs
98b6cf4ac8ddbba7687434d7cc085c4dd84bbcf4 @dain #12713 ✅ rn ✅ docs
44f0138f883bbc79c5f9d2c7b8fd6d7e1fbf5489 @dain #12713 ✅ rn ✅ docs
474b05208b8e43ccd88df5e6c885310c6c6b0d3c @dain #12713 ✅ rn ✅ docs
23063bd9b46f9ef73c1789e6980f669e5ed08aa8 @dain #12713 ✅ rn ✅ docs
94766dc9f726b8280abe960001d7a532c8979c9d @dain #12713 ✅ rn ✅ docs
ef224bbdd33e9447a24462aa12ae9e7e70a9d6ed @dain #12713 ✅ rn ✅ docs
27ef01e615d0fa6529e3917250207e5c446d213f @dain #12713 ✅ rn ✅ docs
329786a923e84be8ff819afe089ab81f5ecb7937 @dain #12713 ✅ rn ✅ docs
7b2cc569e6bd584e91d52f91f687be3f2e083478 @ebyhr #12751 ✅ rn ✅ docs

## 10 June 2022 ✅ 

45394ba5ba11613ba3bd79ae4f2ba86f194562a9 @dain #12780 ✅ rn ✅ docs
802f4f72b33621f5a1463b082e2d7c6e9e7200dc @findepi #12720 ✅ rn ✅ docs
d0d5c6b0c8f5eb7e636096f873561c27e42abe35 @findepi #12720 ✅ rn ✅ docs
4c07efc8a44b14af92fbd19aec5cb0a6a63f168c @ebyhr #12784 ✅ rn ✅ docs
7b0af228aa41bbbe0346473a96ab0a8b161ae5e3 @arhimondr #12645 ✅ rn ✅ docs
df3da06e9951376134a27536a947b40d0e6842b6 @findepi #12658 ✅ rn ✅ docs
0e2a2a42ec259dce724a7ceb6f70556e4cd88885 @findepi #12658 ✅ rn ✅ docs
a2a7eceb93fdf7e290d52e34c9468e7f87c07ab0 @findepi #12658 ✅ rn ✅ docs
daf346dd08dc934271569d4b61aa208c8b0ac23b @findepi #12658 ✅ rn ✅ docs
14baf10a94dde1f2414bd3b934160129e118bb82 @findepi #12658 ✅ rn ✅ docs
f94ea19ee19aaf951693da7630b7dc53dc80977c @findepi #12658 ✅ rn ✅ docs
ad98d554f26175d492b8922bc22b24832400c27a @electrum #12593 ✅ rn ✅ docs
c0e5e97decb6a80d2496ebc42f18aaf9073dd4b7 @electrum #12593 ✅ rn ✅ docs
6da8ef55e9d83f3c32b91750bf67be5e088e9c88 @electrum #11905 ✅ rn ✅ docs
041b8dda08cbed69f711d0173a9a1a7096b9f270 @electrum #12777 ✅ rn ✅ docs

## 11 June 2022 ✅ 

682f5edb3e5ffe61700d68212faebaa7e3143409 @martint (no PR?) ✅ rn ✅ docs
83e144a825a2ee604cdba1ee34baffcbbd37388c @findepi #12788 ✅ rn ✅ docs
bc1c76750f8dc1516466f48bb9eaf983936a3d5a @findepi #12788 ✅ rn ✅ docs

## 12 June 2022 ✅ 

cbe2dca81bc2fbd059c104c59079b939f56908a9 @electrum #12388 ✅ rn ✅ docs
c293a4619bca0243b1f8eeadfd48fe59434dbd27 @electrum #12388 ✅ rn ✅ docs

## 13 June 2022 ✅ 

f247b89be1e981b90edc51d34975e03a05356c17 @findepi #12790 ✅ rn ✅ docs
27c2bf2ff2ecb39fb04ed3940bd1261849f16174 @findepi #12774 ✅ rn ✅ docs
144de287622ff449f78ef3a7ce965055b0c22f55 @findepi #12774 ✅ rn ✅ docs
2118ac5e963c6587c02054dd0e7a8c616614d6d1 @findepi #12774 ✅ rn ✅ docs
0f68de8adbac44e076288e7c84f3a95bc17ce2cc @findepi #12787 ✅ rn ✅ docs
5af3adb4b79a6869dbe40e806abef4ce5258ff32 @findepi #12402 ✅ rn ✅ docs
b8faa60dca27342723e75aea9dd729a139886fa6 @kasiafi #12813 ✅ rn ✅ docs
e071da402efb4b95f6c8b13b2a241e86e446e2a8 @phd3 #12520 ✅ rn ✅ docs
fa83b65b2cee74bf4041d57f992b1a914331715e @ebyhr #12621 ✅ rn ✅ docs
f2cee44110e15df560933a0d47e668f4217f533d @martint #12804 ✅ rn ✅ docs
c5dbf40ef2cedff88813374c9cf6f76bf107900f @ebyhr #12825 ✅ rn ✅ docs
7ede38991ecde1c988fd57be4bc356d315c3b7e9 @ebyhr #12824 ✅ rn ✅ docs

## 14 June 2022 ✅

c84b2c0b636a1cac8c73649819862cb2c404d63c @findepi #12816 ✅ rn ✅ docs
f094c62b4fe709e3470ed211fc5ac1ce5664786d @findepi #12808 ✅ rn ✅ docs
3247bd2e64d7422bd13e805cd67cfca3fa8ba520 @findepi #12808 ✅ rn ✅ docs
8de0cb68850486ad63e66fffe5d918c2374a0c7e @ebyhr #12755 ✅ rn ✅ docs
a3a373f99dfe4a141fcfcb86a6e7bea841d27a4f @hashhar #12608 ✅ rn ✅ docs
3d2e72eb0cfc15d836a36cbf356373b4ec475e01 @ebyhr #12807 ✅ rn ✅ docs
45065c84147aa0fb6b97b6e0d019f4bdd780421f @findepi #12792 ✅ rn ✅ docs
c19dfc1ad19322d5583c7dbda63830f873cb101d @findepi #12792 ✅ rn ✅ docs
3f669704fa33323c064ce24487b1dd67e733043c @findepi #12542 ✅ rn ✅ docs
c51be504125625a70240a7a196de364a02894394 @findepi #12542 ✅ rn ✅ docs
2a466edc37e26047d8cbc7f98023aa62452e782c @findepi #12542 ✅ rn ✅ docs
1f8f92ca425329486d6f7da7243c41305384dd11 @ebyhr #12820 ✅ rn ✅ docs
1628d1d19778236e55e4cb921dded2eaf1a452e3 @martint #12772 ✅ rn ✅ docs
6b07455e70fd20df70f23a960a0d5a73dd05e071 @martint #12693 ✅ rn ✅ docs
1be4a8cdd9806b23dd0dd466e5390bcf8cff7885 @findepi #12836 ✅ rn ✅ docs
ad5253fd96631592a981f1d0d1db6199054605ed @losipiuk #12835 ✅ rn ✅ docs
55c008bb8783b41aa38808671dc30ec57e96773f @ebyhr #12391 ✅ rn ✅ docs

## 15 June 2022

efd264ffa14f70f284fe246f85bdca115c89da71 @findepi #12837 ✅ rn ✅ docs
7aa4a0ec6c4650bb7681466feebb1cc3121fbfee @findepi #12837 ✅ rn ✅ docs
312d2acd08b5925792408bad63fe724476934237 @findepi #12849 ✅ rn ✅ docs
5dbbe7d8f8120551e5ce42e78b7e7371bffde6a8 @findepi #12855 ✅ rn ✅ docs
0c19f3e24484af0af1b08c4062a90942650e64d3 @martint #12830 ✅ rn ✅ docs
369b7c9f7e7787091bc402e26d236eb41e0ab510 @martint #12768 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-386.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* Fix `ClassNotFoundException: Class io.trino.plugin.hive.s3.TrinoS3FileSystem`
+  error when querying `information_schema.columns`. ({issue}`12676`)
+* Fix unnecessarily creating a table snapshot when a write statement does not
+  change the table state. ({issue}`12319`)
+* Fix incorrect query results when filtering on `$path` synthetic column and on
+  at least one other column. ({issue}`12790`)
+* Fix potential query failure when reading `timestamp(6) with time zone` values. ({issue}`12804`)
+* Fix query failure when using the `[VERSION | TIMESTAMP] AS OF` clause on a
+  table with redirection. ({issue}`12542`)
+* Fix query failure when reading a `timestamp with time zone` value before 1970
```
Comment (martint) [inline docs/src/main/sphinx/release/release-386.md:49]: `timestamp(p) with time zone`

File: docs/src/main/sphinx/release/release-386.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+## Accumulo connector
+
+* Disallow creating a view in a non-existent schema. ({issue}`12475`)
+
+## Delta Lake connector
+
+* Improve query performance and scheduling on tables with many small files. ({issue}`12755`)
+* Disallow reading tables if `delta.columnMapping.mode` table property is
+  specified. ({issue}`12621`)
+* Add target maximum file size with 1 GB default. The target max file
```
Comment (martint) [inline docs/src/main/sphinx/release/release-386.md:25]: > Set a target maximum file size during table writes. The default is 1 GB and can be configured via the ...

File: docs/src/main/sphinx/release/release-386.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  size can be set with the `target_max_file_size` session property or the
+  `target-max-file-size` configuration property. ({issue}`12820`)
+
+## Hive connector
+
+* Fix Parquet incompatibility with Apache Hive when writing low-precision
```
Comment (martint) [inline docs/src/main/sphinx/release/release-386.md:31]: "low-precision" doesn't mean anything. Let's be more specific and say "with precision smaller than X".

Also, I'd rephrase as:

> Fix incompatibility with Apache Hive when writing decimal values with precision smaller than X with the experimental Parquet writer.

File: docs/src/main/sphinx/release/release-386.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+## General
+
+* Improve fault-tolerant execution performance by updating default configuration
+  values. ({issue}`12646`)
+* Improve exchange spooling file enumeration speed. ({issue}`12615`)
+* Remove support for Phoenix 4. ({issue}`12772`)
+
+## JDBC driver
+
+* Add configuration property `assumeLiteralUnderscoreInMetadataCallsForNonConformingClients`
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-386.md:12]: hahah .. thats a terrible choice of a property name

Comment (mosabua) [reply docs/src/main/sphinx/release/release-386.md:12]: also .. do we need to make this more user facing .. or is this purely a developer facing property .. and do we have docs for this?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-386.md (hunk 4 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-386.md:12]: It does have documentation, yeah. They had to make the table wider to fit the property in, but they did it! 😆

File: docs/src/main/sphinx/release/release-386.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# Release 386 (15 June 2022)
+
+## General
+
+* Improve fault-tolerant execution performance by updating default configuration
+  values. ({issue}`12646`)
+* Improve exchange spooling file enumeration speed. ({issue}`12615`)
```
[the remark this answers is not in the document]
Comment (colebow) [reply docs/src/main/sphinx/release/release-386.md:7]: I thought this and the comment below were both addressing the upper note, whoops.
